JSONObject.java

```diff
@@ -578,10 +581,10 @@ public double getDouble(String key) throws JSONException {
         Object object = this.get(key);
         try {
             return object instanceof Number ? ((Number) object).doubleValue()
-                    : Double.parseDouble((String) object);
+                    : new BigDecimal((String) object).doubleValue();
```

Comment:
* (int)Double.parseDouble(x)
* (long)Double.parseDouble(x)

do not perform the same as `new BigDecimal(x).*Value` for values larger than 2^53.